tetris.py

```diff
@@ -220,81 +225,113 @@ def draw_block(self):
+
+    def new_block(self):
+        if self.game_status == GameStatus.in_progress:
+            self.current_block_center = (int(game_width / 2), -2)
+            self.block_letter = self.upcoming_block_letter
+            self.upcoming_block_letter = random.choice(block_letters)
+            self.small_board.draw_block(self.get_upcoming_block_shape(), self.upcoming_block_letter, self.landed_blocks)
+            self.rotate_counter = 0
+            self.fast_down = False
 
     def get_block_shape(self):
         (x, y) = self.current_block_center
-        if self.block_letter == "I":
-            coords = [[(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]]
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
+        coords = [self.get_coords(self.block_letter, x, y)]
 
-        if self.block_letter == "L":
-            coords = [[(x - 1, y), (x, y), (x + 1, y), (x + 1, y - 1)]]
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
+        if self.block_letter == "I" or self.block_letter == "Z" or self.block_letter == "Z_rev":
             coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
 
-        if self.block_letter == "L_rev":
-            coords = [[(x - 1, y - 1), (x - 1, y), (x, y), (x + 1, y)]]
+        if self.block_letter == "L" or self.block_letter == "L_rev" or self.block_letter == "E":
             coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
             coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
             coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
 
-        if self.block_letter == "O":
-            coords = [[(x - 1, y), (x, y), (x, y - 1), (x - 1, y - 1)]]
+        return coords[self.rotate_counter % len(coords)]
 
-        if self.block_letter == "E":
-            coords = [[(x - 1, y), (x, y), (x + 1, y), (x, y - 1)]]
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
+    def get_upcoming_block_shape(self):
+        if self.upcoming_block_letter == 'I':
+            self.upcoming_block_center = (2, 0)
+        else:
+            self.upcoming_block_center = (1, 1)
+        (x, y) = self.upcoming_block_center
 
-        if self.block_letter == "Z":
-            coords = [[(x - 1, y - 1), (x, y - 1), (x, y), (x + 1, y)]]
-            coords.append([rotate_point(point, self.current_block_center) for point in coords[-1]])
+        coords = self.get_coords(self.upcoming_block_letter, x, y)
+        return coords
```

Comment:
```suggestion
        return self.get_coords(self.upcoming_block_letter, x, y)
```

Reply:
Resolved as part of get_block_shape change